### PR TITLE
Issue 86: Address Incompatible BuildID & Container Stop

### PR DESCRIPTION
**Pull request by ElusiveMind**
This addresses Issue #86 and additionally addresses another issue which strangely caused completed, passed docker containers to be shut down at the end of the build process.

### CODE REVIEW

File: lib/container_manager/task_runner.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ module.exports = function* runTasks(tasks, opts) {
   }
 
   var taskError = null;
+  var allStepsCompleted = false;
```
Comment (lliss) [inline lib/container_manager/task_runner.js]: Are these changes still necessary?

File: lib/ContainerManager.js [hunk 2]
```diff
@@ -1055,10 +1056,19 @@ Server.prototype.storeBuildData = function(data, done) {
...
 
-
 Server.prototype.storeInDB = function(key, data, done) {
   if (!data.id) {
-    data.id = format(this.flakeIdGen.next(), 'hex');
+    // If we do not have a data id then we need to produce one
+    // that is compatible with the ones generated by ProboCI.
+    // We will use the Crypto library to generate our groups
+    // of random hex numbers.
+    data.id =
```
Comment (lliss) [inline lib/ContainerManager.js]: The format that matches the regex is a uuid (v4). Instead of faking it with random data and building it yourself you could just return a uuid via 
https://www.npmjs.com/package/uuid.

Comment (ElusiveMind) [inline lib/ContainerManager.js]: Great catch. I'll make the adjustment.

**Comment by ElusiveMind** [comment]
Changes made and re-submitted.
